lib/jekyll/readers/layout_reader.rb

```diff
@@ -7,23 +7,30 @@ def initialize(site)
     end
 
     def read
-      layout_entries.each do |f|
-        @layouts[layout_name(f)] = Layout.new(site, layout_directory, f)
+      layout_entries.each do |directory, entries|
+        entries.each do |name|
+          @layouts[layout_name(name)] = Layout.new(site, directory, name)
+        end
       end
 
       @layouts
     end
 
-    def layout_directory
-      @layout_directory ||= (layout_directory_in_cwd || layout_directory_inside_source)
+    def layout_directories
+      @layout_directories ||= Array(site.config['layouts_dir']).map do |directory|
```

Comment:
I am not convinced this should be configurable as an array. I think it's useful for themes, but the `Site` can ask the `Theme` for its directory. I'd rather see this as:

```ruby
def layout_directories
  @layout_directories ||= Array(site.config['layout_dir']) + Array(theme.layout_dir)
end
```

Before #4595, just the first piece will do.